students/Z_shen/lesson03/mailroom.py

```diff
@@ -0,0 +1,100 @@
+
+donor = [
+    ["William Gates", 653772.32, 2],
+    ["Jeff Bezos", 16396.10, 3],
+    ["Paul Allen", 877.33, 1],
+    ["Mark Zuckerberg",708.42, 3],
+]
+
+def donor_name_list(donor):
+    donor_name=[]
+    for name in donor:
+        donor_name.append(name[0])
+    return donor_name
+
+def print_donor_report(donor):
+    row='| {name:20} | {sign_1:1}  {amount:>15} | {num_gifts:^10} | {sign_2:1} {avg_gift:<15,.2f} |'.format
+
+    print('| {:20} | {:1}  {:^15} | {:^10} | {:1} {:<15} |'.format('Donor Name', ' ' ,'Total Given','Num Gifts','','Average Gift'))
+    print('-'*78)
+    for p in donor:
+        print(row(name=p[0],sign_1='$',amount=p[1], num_gifts=p[2],sign_2='$',avg_gift=int(p[1])/int(p[2])))
```

Comment:
Donors should be listed by total given in descending order, including new donors and donations